modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/capability/xml/description/ParameterDescriptionDocumenter.java

```diff
@@ -44,7 +45,8 @@ void document(ParameterizedDeclaration<?> parameterized, Element method, MethodD
 
     if (method instanceof ExecutableElement) {
       ((ExecutableElement) method).getParameters().stream()
-          .filter(e -> e.getAnnotation(ParameterGroup.class) != null)
+          .filter(e -> e.getAnnotation(ParameterGroup.class) != null
```

Comment:
I'm putting this comment here but also take this as feedback for the entire PR (and for future Forward compatibility PRs as well). I know this is how we've been doing it in the past (me included) but I think that this pattern of having if clauses with `||` operators checking for one annotation or the other, or having collections in which we aggregate annotated components, followed by and `addAll`/`putAll` doing the same for another annotation, is getting totally out of hand and really affecting the code's readability. 

Please extract all of those things to other methods (either private in the same class, or at an utils class, doesn't matter) so that instead of having `if (isAnnotatedWith(X) or || isAnnotatedWithY)` we can have `if (someSemanticIsPresent()` instead. I think it's really important we keep syntax and semantics decoupled.